### DISCUSSION TITLE
No valid membership found --> but user is member

### DISCUSSION THREAD

**Discussion by orcofthedarkness** [liedekef/events-made-easy#223] [Q&A] [answered]
Hi there,

running v. 2.3.70 

I have an event where only a sub-group of users are allowed to book a seat. So i created a membership = UCI. Then i added a bunch of users to the very same membership. 
For testing purpose i configured the follwoing "checks" in the event template:
```
[eme_if tag='#_IS_USER_MEMBER_OF{UCI}' value='0'] NO [/eme_if]
[eme_if tag='#_IS_USER_MEMBER_OF{UCI}' value='1'] YES [/eme_if]
```
which works just fine

For the event itself i created a new category = eventuci
and i added these lines to the event template:
```
[eme_if tag='#ESC_CATEGORIES' value='eventuci']</br>#_ADDBOOKINGFORM_IF_USER_IS_MEMBER_OF{UCI}[/eme_if]
```
and in the RSVP section this is set:
![image](https://github.com/liedekef/events-made-easy/assets/127893294/4880737f-98c6-4420-8994-71291941dea6)

So, if the event is in category = "eventuci" AND the user is member of = "UCI" the booking form will be shown.
which works fine (tested in either configuration - user is/is not member; event is/is not in the correct category)

but, as soon as i want to book the event, i get the following message: "No valid membership found".

Did i miss something? Has anyone an idea what i could possibly check or what i did wrong?
thanks in advance,
Ron

**Comment by liedekef** [comment, marked as answer]
Hum, that's weird. If the "Require EME membership" is checked, normally the booking form is only shown if the user has not passed the max usage count set in the membership. Can you check the max usage count in the membership and also the actual usage count for the member?

**Comment by orcofthedarkness** [comment]
thank you for your quick response.

okay. that "fixed" somehow the issue...
in the membership the "max usage count" was set = 0 --> I thought this is "unlimited"
Increasing the value to something, while haveing the actual usage count below that helped mand booking/cancelling works as expected

is there any way, to set the max usage count to unlimited? otherwise, i'll set the "max usage count" = zillion...
thank you so much!
Ron

**Reply by liedekef** [reply]
normally 0 means unlimited, but maybe you've set it to something else in the past, made some tests and switched to 0 later on?
Anyway, I'll check the code again to make sure.

**Reply by liedekef** [reply]
This should help: https://github.com/liedekef/events-made-easy/commit/8c916c0f0b2d5804e9e50e36ce4957f3d45d7088

**Reply by liedekef** [reply]
Ok, that fix was not enough. This extra change should help:
https://github.com/liedekef/events-made-easy/commit/3a0deb7396d86a173afe227034079b6e31a0b441

**Comment by orcofthedarkness** [comment]
Hi again,

applied the fixes, tested them and happy to report that it works now with unlimited (=0) maximum usage count

Thank you very much!
Ron
